posthog/queries/base.py

```diff
@@ -381,3 +398,28 @@ def is_truthy_or_falsy_property_value(value: Any) -> bool:
         or value is True
         or value is False
     )
+
+
+def relative_date_parse_for_feature_flag_matching(value: str) -> Optional[datetime.datetime]:
+    regex = r"(?P<number>[0-9]+)(?P<type>[a-z])"
+    match = re.search(regex, value)
+    parsed_dt = datetime.datetime.now(tz=ZoneInfo("UTC"))
+    if match:
+        number = int(match.group("number"))
+        type = match.group("type")
```

Comment:
This will overshadow the built-in _type()_ function, maybe consider a different name?